.github/workflows/mcad-release.yml

```
@@ -0,0 +1,43 @@
+# This workflow will create a GitHub release tag
+
+name: mcad Release
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Release tag'
+        required: true
+        default: 'v0.0.0-dev'
+        type: string
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout code
+      uses: actions/checkout@v3
+
+    - name: Configure Git
+      run: |
+        git config user.name "GitHub Actions"
+        git config user.email "github-actions@github.com"
+
+    - name: check tag format
+      run: |
+        if [[ "${{ github.event.inputs.tag }}" =~ "^v\d+\.\d+\.\d+$" ]]; then
```

`\d` doesn't work in bash, you can use:
```suggestion
        if [[ "${{ github.event.inputs.tag }}" =~ ^v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$ ]]; then
```

---

```suggestion
        if [[ "${{ github.event.inputs.tag }}" =~ ^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$" ]]; then
```

From https://semver.org/ at the bottom of the page

This covers pre-release tags

---

I dont think so we will do pre-release tags..changing to **^v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$**